### PR TITLE
fix: Current generative drops are not unlockable

### LINKED ISSUE #7843: Current generative drops are not unlockable
And I think generative drops are currently not unlockables.

image

ref
---
- #7838 
- #7580

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

  👇 __ Let's make a quick check before the contribution.

  ## PR Type

  - [x] Bugfix
  - [x] Refactoring

  ## Needs QA check

  - @kodadot/qa-guild please review

  ## Context

  - [x] Closes #7843
  - [ ] Requires deployment <snek/rubick/worker>

  #### Did your issue had any of the "$" label on it?

  - [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

  ## Screenshot 📸

  - [ ] My fix has changed UI

  ## Copilot Summary
  
### <samp>🤖 Generated by Copilot at af98ea8</samp>

This pull request refactors the drop components to simplify their structure and extract common elements, such as the unlockable item info, into reusable components. It also introduces a new component `CollectionUnlockableItemInfo` that displays the unlockable content explanation based on the collection id. Additionally, it modifies the `useUnlockable` composable to accept a more generic type for the entity parameter. These changes improve the code readability, maintainability, and flexibility.

  
### <samp>🤖 Generated by Copilot at af98ea8</samp>

> _We are the masters of the code, we break the duplication_
> _We extract the logic, we create the `CollectionUnlockableItemInfo`_
> _We use the `script setup`, we unleash the power of Vue_
> _We refactor the drops, we make them shine like new_

### CODE REVIEW

File: composables/useUnlockable.ts
Comment (Jarsen136) [inline composables/useUnlockable.ts:1]: It looks like the api is checking if **the nft** is the unlockable item. Is there any idea to check if **the collection** is  unlockable? @vikiival 
https://github.com/kodadot/nft-gallery/issues/7843#issuecomment-1783244325
`resolve/${prefix}-${nftId}`

Comment (Jarsen136) [reply composables/useUnlockable.ts:1]: @vikiival Could you please take a look at it?

Comment (vikiival) [reply composables/useUnlockable.ts:1]: > @vikiival Could you please take a look at it?

You have been pranked by bad naming


Usage of keywise is 

```
const { isUnlockable, unlockLink } = useUnlockable(collection)
```

Therefore it is checking as 

```
GET /resolve/${prefix}-${collectionId}
```

**Comment by prury** [comment]
still no unlock-able button showing here for me: 
image

**Comment by prury** [comment]
what about this section? should we hide it for non unlockables as well? 
image

it seems now that no drop is showing this part: 

image

as far as i know waifu drops should have and generative and kodachain should not

**Comment by Jarsen136** [comment]
> what about this section? should we hide it for non unlockables as well? image
> 
> it seems now that no drop is showing this part:
> 
> image
> 
> as far as i know waifu drops should have and generative and kodachain should not

You're right. I have fixed them.

**Comment by prury** [comment]
last thing, since you've removed the unlockable section, kodachain drop needs some spacing between the footer
image

**Comment by Jarsen136** [comment]
> last thing, since you've removed the unlockable section, kodachain drop needs some spacing between the footer image

✅ Added some spacing between them.

**Comment by vikiival** [comment]
> kodachain drop needs some spacing between the footer

KodaChain should be unlockable 🤔

**Comment by prury** [comment]
> > kodachain drop needs some spacing between the footer
> 
> KodaChain should be unlockable 🤔

what it should unlock? because for the waifu mints we unlock $ vouchers, but the kodachain we just give ppl the nft in case they voted for proposal

**Comment by yangwao** [comment]
Thanks!
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 4.34 USD/DOT ~ 6.912 $DOT
🧗 [16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z](https://polkadot.subscan.io/account/16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)
🔗 [0x73a75117f4db8167c2fd67ac4725a369f97f671046d2d9f9b40520ddf0ba5ed4](https://polkadot.subscan.io/extrinsic/0x73a75117f4db8167c2fd67ac4725a369f97f671046d2d9f9b40520ddf0ba5ed4)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by yangwao** [comment]
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 4.35 USD/DOT ~ 6.897 $DOT
🧗 [16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z](https://polkadot.subscan.io/account/16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)
🔗 [0x8393c6a8c499b02ca0001fc6c275c84801c776075031b5a48f1267514067c038](https://polkadot.subscan.io/extrinsic/0x8393c6a8c499b02ca0001fc6c275c84801c776075031b5a48f1267514067c038)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**